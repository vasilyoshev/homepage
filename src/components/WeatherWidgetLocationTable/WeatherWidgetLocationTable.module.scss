.alert {
  width: 150px;
}

.cityTableCell {
  width: 150px;
}

.tableContainer {
  width: 400px;
}
