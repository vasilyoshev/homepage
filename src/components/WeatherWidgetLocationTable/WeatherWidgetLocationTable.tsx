import React from 'react';
import { WeatherWidgetLocationTableProps } from 'interfaces';
import { Table, TableContainer, TableBody, TableRow, TableCell } from '@material-ui/core';
import Alert from '@material-ui/lab/Alert';

export const WeatherWidgetLocationTable: React.FC<WeatherWidgetLocationTableProps> = ({ locations, onSelect, current }) => {
  return (
    <TableContainer>
      <Table size="small" aria-label="a dense table">
        <TableBody>
          {locations.map((location) =>
            <TableRow key={location.id} onClick={() => onSelect(location)}>
              <TableCell>{location.name}</TableCell>
              <TableCell> {current?.id === location.id ? <Alert severity="info">Current City</Alert> : ''}</TableCell>
            </TableRow>,
          )}
        </TableBody>
      </Table>
    </TableContainer>
  );
};
