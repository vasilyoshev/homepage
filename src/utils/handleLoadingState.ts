import axios from 'axios';
import { decrementLoadingCount, incrementLoadingCount } from 'slices';
import store from 'store';

export const handleLoadingState = (): void => {
  axios.interceptors.response.use(
    (response) => {
      setTimeout(() => store.dispatch(decrementLoadingCount()), 200);

      return response;
    },
    () => {
      store.dispatch(decrementLoadingCount());
    },
  );

  axios.interceptors.request.use((request) => {
    store.dispatch(incrementLoadingCount());

    return request;
  });
};
