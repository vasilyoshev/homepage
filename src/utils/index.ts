export * from './handleLoadingState';
export * from './testUtils';
