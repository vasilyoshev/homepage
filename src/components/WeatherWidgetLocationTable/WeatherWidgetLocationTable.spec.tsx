import { render } from 'utils';
import { WeatherWidgetLocationTable } from './WeatherWidgetLocationTable';

describe('WeatherWidgetLocationTable', () => {
  it('renders successfully', () => {
    const { asFragment } = render(<WeatherWidgetLocationTable locations={[]} onSelect={jest.fn()} current={null} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
