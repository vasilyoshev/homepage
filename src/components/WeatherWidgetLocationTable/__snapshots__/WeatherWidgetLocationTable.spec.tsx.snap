// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherWidgetLocationTable renders successfully 1`] = `
<DocumentFragment>
  <div
    class="MuiTableContainer-root"
  >
    <table
      class="MuiTable-root"
    >
      <tbody
        class="MuiTableBody-root"
      />
    </table>
  </div>
</DocumentFragment>
`;
